### PR TITLE
feat: column

### PR DESCRIPTION
**Pull request by michalsn**
This PR adds the `column` method.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -237,6 +237,32 @@ return $collection->average('foo');
...
+$collection = new Collection([
+    ['id' => 1, 'name' => 'John'],
+    ['id' => 2, 'name' => 'Carter'],
+    ['id' => 3, 'name' => 'Steve'],
+]);
+return $collection->column('name');
+// return ['John', 'Carter', 'Steve']
+```
+
+If an index is provided, the values from this column will be used as keys for the returned result in the collection.
```
Comment (lonnieezell) [inline docs/index.md:254]: Nice feature!

**Review by lonnieezell — CHANGES_REQUESTED** [review]
Could we make this work with objects also? I can see this being very handy for Entities. Laravel has something similar called `pluck` that I use quite frequently. 

Basically, if its an object we loop over it and create an array element with a key for the index and the value of the public property (if exists).

**Comment by michalsn** [comment]
Hmm... this method already handles objects. And thanks to the magic methods in the Entity class it "just works". I may missed something here though.

Are you thinking of any specific scenario? I looked at the `pluck` method and the only difference is that it also handles nested values.

**Review by lonnieezell — APPROVED** [review]

**Comment by lonnieezell** [comment]
> Hmm... this method already handles objects. And thanks to the magic methods in the Entity class it "just works".

Nope! All good. I completely spaced that `array_column` worked with objects. Approved and merging. Thanks!

**Pull request merged**